Library/Homebrew/formula.rb

```diff
@@ -2393,8 +2292,87 @@ def to_hash
   end
 
   # @private
-  def to_hash_with_variations
-    hash = to_hash
+  def to_api_hash
+    api_hash = {
+      "desc"                 => desc,
+      "license"              => SPDX.license_expression_to_string(license),
+      "homepage"             => homepage,
+      "urls"                 => urls_hash.transform_values(&:compact),
+      "post_install_defined" => post_install_defined?,
+      "ruby_source_path"     => ruby_source_path,
+      "ruby_source_checksum" => { "sha256" => ruby_source_checksum&.hexdigest },
+    }
+
+    dep_hash = dependencies_hash
+               .except("recommended_dependencies", "optional_dependencies")
+               .transform_values(&:presence)
+               .compact
+
+    api_hash.merge(dep_hash)
+
+    # Exclude default values.
+    api_hash["revision"] = revision unless revision.zero?
+    api_hash["version_scheme"] = version_scheme unless version_scheme.zero?
+
+    api_hash["key_only_reason"] = keg_only_reason.to_hash if keg_only_reason
+    api_hash["pour_bottle_only_if"] = self.class.pour_bottle_only_if.to_s if self.class.pour_bottle_only_if
+    api_hash["link_overwrite"] = self.class.link_overwrite_paths.to_a if self.class.link_overwrite_paths.present?
+
+    if stable
+      api_hash["versions"] = { "stable" => stable&.version&.to_s }
+      api_hash["bottle"] = { "stable" => bottle_hash(compact_for_api: true) } if bottle_defined?
```

Comment:
Don't think the `stable` nesting makes sense any more; it existed originally because we also had `devel` here and because we considered we might have `devel` bottles.

Reply:
Good point, I'll remove it. It will just be `version` and `bottle` going forward.

Reply:
Addressed in https://github.com/Homebrew/brew/pull/16541/commits/3a70b6aeae363c66a6d35f8d51397350aacae35a